cupy/cuda/device.pyx

```diff
@@ -247,6 +252,35 @@ cdef class Device:
         """
         return _get_attributes(self.id)
 
+    @property
+    def pci_bus_id(self):
+        """A string of the PCI Bus ID
+
+        Returns:
+            pci_bus_id (str):
+                Returned identifier string for the device in the following
+                format [domain]:[bus]:[device].[function] where domain, bus,
+                device, and function are all hexadecimal values.
+        """
+        return _get_pci_bus_id(self.id)
+
+    @classmethod
+    def from_pci_bus_id(cls, pci_bus_id):
+        """Returns a new device instance based on a PCI Bus ID
+
+        Args:
+            pci_bus_id (str):
+                The string for a device in the following format
+                [domain]:[bus]:[device].[function] where domain, bus, device,
+                and function are all hexadecimal values.
+        Returns:
+            device (Device):
+                An instance of the Device class that has the PCI Bus ID as
+                given by the argument pci_bus_id.
+        """
+        new_id = runtime.deviceGetByPCIBusId(str(pci_bus_id))
```

Comment:
A nitpick but "new" implies creating a new entity, but the device ID is persistent for a particular physical device.

```suggestion
        device_id = runtime.deviceGetByPCIBusId(str(pci_bus_id))
```